File: practice/leetcode/python/242_valid_anagrams.py
#https://leetcode.com/problems/valid-anagram/description/

class Solution:
    def validAnagram(self, s: str, t: str) -> bool:
        s_map = {}

        if len(s) != len(t):
            return False

        for c in s:
            s_map[c] += 1

        for c in t:
            if not c in s_map or s_map[c] == 0:
                return False
            s_map[c] -= 1

        return True

File: practice/leetcode/cpp/242_valid_anagrams.cpp
//https://leetcode.com/problems/valid-anagram/description/

#include <algorithm>
#include <string>
#include <vector>
using namespace std;

class Solution {
public:
	bool isAnagram(string s, string t) {
		vector<int> charCount(26, 0);
		if (s.length() != t.length()) {
			return false;
		}
		for (int i = 0; i < s.length(); i++) {
			charCount[s[i] - 'a']++;
			charCount[t[i] - 'a']--;
		}
		return all_of(charCount.begin(), charCount.end(), [] (int x) { return x == 0;});
	} 
};
